crates/types/src/schema/mod.rs

```diff
@@ -110,6 +112,107 @@ pub mod storage {
     flexbuffers_storage_encode_decode!(Schema);
 }
 
+mod serde_hacks {
+    use super::*;
+    use crate::identifiers;
+
+    /// The schema information
+    #[serde_as]
+    #[derive(Debug, Clone, serde::Serialize, serde::Deserialize)]
+    pub struct Schema {
+        // --- Old data structure
+        #[serde(default, skip_serializing_if = "Option::is_none")]
+        pub services: Option<HashMap<String, ServiceSchemas>>,
+        // flexbuffers only supports string-keyed maps :-( --> so we store it as vector of kv pairs
+        #[serde_as(as = "Option<serde_with::Seq<(_, _)>>")]
+        #[serde(default, skip_serializing_if = "Option::is_none")]
+        pub deployments: Option<HashMap<DeploymentId, DeploymentSchemas>>,
+
+        // --- New data structure
+
+        // Registered deployments
+        #[serde_as(as = "Option<restate_serde_util::MapAsVec>")]
+        #[serde(default, skip_serializing_if = "Option::is_none")]
+        pub deployments_v2: Option<HashMap<DeploymentId, v2::Deployment>>,
+        // Currently selected service revisions to use for new invocations.
+        // This is currently always the last revision, but it might change if we want to allow people manually
+        // selecting the service revision to route requests to (or some A/B testing stuff)
+        #[serde(default, skip_serializing_if = "Option::is_none")]
+        pub current_service_revisions: Option<HashMap<String, identifiers::ServiceRevision>>,
```

Comment:
This seems unrelated to the overall changes. Should this be split into a separate PR once requirements are a bit clearer?

Reply:
It's not unrelated, I need it to rematerialize from the new data structure to the old (i need to know which one is the current service revision to rebuild the index).
This is also generally something I need to store somewhere for the reasons described in the comment.

Reply:
After talking offline, we concluded that this field is probably not needed.